File: 🟡 Medium/50 - Pow(x, n)/_50Pow(x, n).py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        ans = 1

        if n < 0:
            x = 1 / x
            n = abs(n)

        while n > 0:
            if (n & 1) == 1:
                ans *= x
            x *= x
            n //= 2

        return ans

File: 🟡 Medium/50 - Pow(x, n)/_50Pow(x, n).java
class Solution {
    public double myPow(double x, int n) {
        double ans = 1.0;
        boolean flag = false;
        long power = n;

        if (n < 0) {
            x = 1 / x;
            power = Math.abs(power);
        }

        while (power > 0) {
            // LSB = 1 (Then multiply your answer with x)
            if ((power & 1) != 0) {
                ans *= x;
            }

            x *= x;
            power = power >> 1;
        }

        return ans;
    }
}